src/react/sebs_snacks/src/components/recipePage.jsx

```diff
@@ -0,0 +1,131 @@
+                                    <h1 id="recipeTitle">{recipe.title}</h1>
+                                    <h4 id="recipeDesc">{recipe.desc}</h4>
+                                    <h2 id="ingredients">Ingredients</h2>
+                                    <div className="recipeIngredients">
+                                        <ul id="recipeIngredientList">
+                                        {recipe.ingredientList.map(function(currentValue, index) { {/* Builds ingredient list display from data*/}
+                                            return <li key={index}>{currentValue}</li>;
+                                            })}
+                                        </ul>
+                                        <h4>Add an Ingredient:</h4>
+                                        <input 
+                                            id="newIngredient" 
+                                            type="text"
+                                            className="form-element" 
+                                            placholder="Ingredient" 
+                                            value={newIngredient} 
+                                            onChange={(e) => {
+                                                setNewIngredient(e.target.value);}} /> {/* Rerenders the page to display the character previously typed */}
+                                        <button onClick={() => {addIngredient(recipe, newIngredient); setNewIngredient('')}}>Add ingredient to list</button> {/*Adds ingredient to lists, and clears input text field, updating the state, forcing the component to rerender*/}
+                                    </div>
+
+                                    <h2 id="method">Instructions</h2>
+                                    <div className="recipeMethod">
+                                        <ol id="recipeInstructionList">
+                                        {recipe.instructionList.map(function(currentValue, index) {
+                                            return <li key={index}>{currentValue}</li>;
+                                            })}
+                                        </ol>
+                                        <h4>Add an Instruction:</h4>
+                                        <input 
+                                            id="newInstruction" 
+                                            type="text"
+                                            className="form-element" 
+                                            placholder="Instruction" 
+                                            value={newInstruction} 
+                                            onChange={(e) => {
+                                                setNewInstruction(e.target.value);}} />
+                                        <button onClick={() => {addInstruction(recipe, newInstruction); setNewInstruction('')}}>Add instruction to list</button>
```

Comment:
to make things a little cleaner, just put your addInstruction function call here, and then reset your setNewInstruction(''), in your actual addInstruction function